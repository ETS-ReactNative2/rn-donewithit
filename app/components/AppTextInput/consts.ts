import { KeyboardTypeOptions } from 'react-native'
import COLORS from '../../config/colors'

type constsType = {
  PLACEHOLDER: string
  ICON_NAME: string
  KEYBOARD_TYPE: KeyboardTypeOptions
  ICON_COLOR: string
  ICON_SIZE: number
  BORDER_RADIUS: number
  PADDING: number
  BACKGROUND_COLOR: string
  MAX_LENGTH: number
  TEXT_INPUT_FONT_SIZE: number
  CLEAR_BUTTON_MODE: string
  HAS_ICON: boolean
  HEIGHT: number
}
const CONSTS: constsType = {
  PLACEHOLDER: 'Write something here...',
  ICON_NAME: 'email',
  KEYBOARD_TYPE: 'default',
  ICON_COLOR: COLORS.GRAY,
  ICON_SIZE: 20,
  BORDER_RADIUS: 25,
  PADDING: 10,
  BACKGROUND_COLOR: COLORS.INPUT_BACKGROUND_COLOR,
  MAX_LENGTH: 50,
  TEXT_INPUT_FONT_SIZE: 18,
  CLEAR_BUTTON_MODE: 'always',
  HAS_ICON: true,
  HEIGHT: 50
}

export default CONSTS
