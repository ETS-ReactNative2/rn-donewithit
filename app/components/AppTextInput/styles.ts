import { Platform, StyleSheet } from 'react-native'
import CONSTS from './consts'
import COLORS from '../../config/colors'

const styles = StyleSheet.create({
  container: {
    backgroundColor: CONSTS.BACKGROUND_COLOR,
    borderRadius: CONSTS.BORDER_RADIUS,
    padding: CONSTS.PADDING,
    height: CONSTS.HEIGHT,
    overflow: 'hidden',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center'
  },
  icon: {
    marginRight: CONSTS.PADDING
  },
  textInput: {
    ...Platform.select({
      android: {
        fontFamily: 'Roboto'
      },
      ios: {
        fontFamily: 'Courier'
      }
    }),
    color: COLORS.BLACK,
    fontSize: CONSTS.TEXT_INPUT_FONT_SIZE,
    flex: 1
  }
})

export default styles
