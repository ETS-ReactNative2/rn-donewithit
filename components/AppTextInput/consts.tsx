import { KeyboardTypeOptions } from 'react-native'
import COLORS from '../../config/colors'

type constsType = {
  PLACEHOLDER: string
  ICON_NAME: string
  KEYBOARD_TYPE: KeyboardTypeOptions
  ICON_COLOR: string
  ICON_SIZE: number
  BORDER_RADIUS: number
  PADDING: number
  BACKGROUND_COLOR: string
  MAX_LENGTH: number
  TEXT_INPUT_FONT_SIZE: number
  CLEAR_BUTTON_MODE: string
}
const CONSTS: constsType = {
  PLACEHOLDER: 'Write something here...',
  ICON_NAME: 'email',
  KEYBOARD_TYPE: 'default',
  ICON_COLOR: COLORS.GRAY,
  ICON_SIZE: 20,
  BORDER_RADIUS: 20,
  PADDING: 10,
  BACKGROUND_COLOR: '#fbf6f8',
  MAX_LENGTH: 50,
  TEXT_INPUT_FONT_SIZE: 18,
  CLEAR_BUTTON_MODE: 'always'
}

export default CONSTS
