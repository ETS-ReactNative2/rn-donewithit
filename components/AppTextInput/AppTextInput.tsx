import React from 'react'
import { TextInput, View } from 'react-native'
import COLORS from '../../config/colors'
import AppIcon from '../AppIcon'
import CONSTS from './consts'
import styles from './styles'

// type AppTextInputProps = {
//   placeholder?: string
//   keyboardType?: KeyboardTypeOptions
//   iconName?: string
//   iconColor?: string
//   iconSize?: number
//   maxLength?: number
//   secureTextEntry?: boolean
//   onChangeText?: (text: string) => void
// }

function AppTextInput({
  // Container
  icon = CONSTS.HAS_ICON,
  style,
  textInputStyle,
  // AppIcon
  iconName = CONSTS.ICON_NAME,
  iconColor = CONSTS.ICON_COLOR,
  iconSize = CONSTS.ICON_SIZE,
  // InputText
  placeholder = CONSTS.PLACEHOLDER,
  keyboardType = CONSTS.KEYBOARD_TYPE,
  maxLength = CONSTS.MAX_LENGTH,
  clearButtonMode = CONSTS.CLEAR_BUTTON_MODE,
  ...otherProps
}: any) {
  return (
    <View style={[style, styles.container]}>
      {icon && (
        <AppIcon
          name={iconName}
          color={iconColor}
          size={iconSize}
          style={styles.icon}
        />
      )}
      <TextInput
        placeholder={placeholder}
        clearButtonMode={clearButtonMode}
        maxLength={maxLength}
        keyboardType={keyboardType}
        style={[styles.textInput, textInputStyle]}
        placeholderTextColor={COLORS.SILVER}
        {...otherProps}
      />
    </View>
  )
}

export default AppTextInput
