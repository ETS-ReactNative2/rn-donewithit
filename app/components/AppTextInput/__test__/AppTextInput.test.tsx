import React from 'react'
import { StyleSheet } from 'react-native'
import renderer from 'react-test-renderer'
import AppTextInput from '../index'

const customStyles = StyleSheet.create({
  textInput: {
    width: 180
  }
})

const props = {
  icon: false,
  placeholder: 'Price',
  keyboardType: 'numeric',
  textContentType: 'none',
  value: '999',
  style: customStyles.textInput
}

describe('AppTextInput', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppTextInput {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
