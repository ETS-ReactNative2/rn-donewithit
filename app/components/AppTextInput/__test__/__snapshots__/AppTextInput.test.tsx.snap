// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTextInput renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "width": 180,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#f9f2f2",
        "borderRadius": 25,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "flex-start",
        "overflow": "hidden",
        "padding": 10,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    clearButtonMode="always"
    keyboardType="numeric"
    maxLength={50}
    placeholder="Price"
    placeholderTextColor="#C0C0C0"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "#000",
          "flex": 1,
          "fontFamily": "Courier",
          "fontSize": 18,
        },
        undefined,
      ]
    }
    textContentType="none"
    underlineColorAndroid="transparent"
    value="999"
  />
</View>
`;
